packages/legend-query-builder/src/components/result/tds/QueryBuilderTDSGridResult.tsx

```diff
@@ -549,12 +552,20 @@ export const QueryBuilderTDSGridResult = observer(
               onRowDataUpdated={(params) => {
                 params.api.refreshCells({ force: true });
               }}
-              onRangeSelectionChanged={(event) => {
+              onCellSelectionChanged={(
+                event: DataGridCellSelectionChangedEvent<QueryBuilderTDSRowDataType>,
+              ) => {
                 const selectedCells = getSelectedCells(event.api);
                 resultState.setSelectedCells([]);
                 selectedCells.forEach((cell) =>
                   resultState.addSelectedCell(cell),
                 );
+                if (
+                  resultState.mousedOverCell === null &&
+                  isNonNullable(selectedCells[0])
+                ) {
+                  resultState.setMouseOverCell(selectedCells[0]);
+                }
```

Comment:
This is the fix for the bug. The `filterByOrOutValues` function call expects `resultState.mousedOverCell` to be set. `resultState.mousedOverCell` gets set on cell click based on `resultState.selectedCells[0]`. Since `resultState.selectedCells[0]` is null on the very first click of the grid, `resultState.mouseOverCell` was getting set to null. This fixes that bug by setting `resultState.mouseOverCell` if it is null when a cell or range is first selected.

Reply:
Went with a different fix so resolving this conversation.